Sources/StreamChat/Repositories/Operations/SyncOperations_Tests.swift

```diff
@@ -0,0 +1,251 @@
+        database = client.mockDatabaseContainer
+        syncRepository = SyncRepositoryMock(client: client)
+    }
+
+    // MARK: - GetChannelIdsOperation
+
+    func test_GetChannelIdsOperation_noChannels() {
+        let context = SyncContext()
+        let operation = GetChannelIdsOperation(database: database, context: context)
+        operation.startAndWaitForCompletion()
+        XCTAssertEqual(context.localChannelIds.count, 0)
+    }
+
+    func test_GetChannelIdsOperation_withChannels() throws {
+        try database.writeSynchronously { session in
+            let query = ChannelListQuery(filter: .exists(.cid))
+            try session.saveChannel(payload: self.dummyPayload(with: .unique, numberOfMessages: 0), query: query)
+        }
+
+        let context = SyncContext()
+        let operation = GetChannelIdsOperation(database: database, context: context)
+        operation.startAndWaitForCompletion()
+        
+        XCTAssertEqual(context.localChannelIds.count, 1)
+    }
+
+    // MARK: - GetPendingConnectionDateOperation
+
+    func test_GetPendingConnectionDateOperation_noUser() {
+        let context = SyncContext()
+        let operation = GetPendingConnectionDateOperation(database: database, context: context)
+
+        operation.startAndWaitForCompletion()
+
+        XCTAssertNil(context.lastPendingConnectionDate)
+    }
+
+    func test_GetPendingConnectionDateOperation_noPendingDate() throws {
+        let context = SyncContext()
+        try database.createCurrentUser()
+        let operation = GetPendingConnectionDateOperation(database: database, context: context)
+
+        operation.startAndWaitForCompletion()
+
+        XCTAssertNil(context.lastPendingConnectionDate)
+    }
+
+    func test_GetPendingConnectionDateOperation_pendingDate() throws {
+        let context = SyncContext()
+        try database.createCurrentUser()
+        let date = Date().addingTimeInterval(-3600)
+        try database.writeSynchronously { session in
+            session.currentUser?.lastPendingConnectionDate = date
+        }
+
+        let operation = GetPendingConnectionDateOperation(database: database, context: context)
+
+        operation.startAndWaitForCompletion()
+
+        XCTAssertEqual(context.lastPendingConnectionDate, date)
+    }
+
+    // MARK: - SyncEventsOperation
+
+    func test_SyncEventsOperation_noPendingDate() {
+        let context = SyncContext()
+        context.lastPendingConnectionDate = nil
+        let operation = SyncEventsOperation(database: database, syncRepository: syncRepository, context: context)
+
+        operation.startAndWaitForCompletion()
+
+        XCTAssertEqual(context.synchedChannelIds.count, 0)
+        XCTAssertTrue("syncMissingEvents(using:channelIds:bumpLastSync:completion:)".wasNotCalled(on: syncRepository))
+    }
+
+    func test_SyncEventsOperation_pendingDate_syncFailure_shouldRetry2times() throws {
```

Comment:
nit: Using values on the names of the tests usually make them outdated pretty fast, in this case, it is already from what I can see the assertion it says "3" times instead of "2"
```suggestion
    func test_SyncEventsOperation_pendingDate_syncFailure_shouldRetry() throws {
```